WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift

```diff
@@ -302,11 +305,31 @@ extension ProductsViewController: UITableViewDelegate {
         ServiceLocator.analytics.track(.productListProductTapped)
 
         let product = resultsController.object(at: indexPath)
-        let currencyCode = CurrencySettings.shared.currencyCode
-        let currency = CurrencySettings.shared.symbol(from: currencyCode)
-        let viewModel = ProductDetailsViewModel(product: product, currency: currency)
-        let productViewController = ProductDetailsViewController(viewModel: viewModel)
-        navigationController?.pushViewController(productViewController, animated: true)
+
+        // TODO: Remove after testing previewing a product
+
+        guard let credentials = ServiceLocator.stores.sessionManager.defaultCredentials,
+            let site = ServiceLocator.stores.sessionManager.defaultSite else {
+            fatalError()
+        }
+
+        let previewGenerator = ProductPreviewGenerator(product: product, frameNonce: site.frameNonce, credentials: credentials)
+        previewGenerator.onFailure = { message in
+            DDLogError("⛔️ Unable to preview product (\(product.name)): \(message)")
+        }
+        previewGenerator.onAttemptURLRequest = { [weak self] request in
+            guard let self = self else {
+                return
+            }
+            WebviewHelper.launch(request.url?.absoluteString, with: self)
+        }
+        previewGenerator.generate()
```

Comment:
API design suggestion:

(Unless I'm not following something!!), if the public usage involves:

1. Setting `onAttemptURLRequest`
2. Calling `generate()`

How about if, instead, `generate` takes a callback closure?

```
previewGenerator.generate { request in
    WebviewHelper.launch(request.url.absoluteString, with: self)
}
```